src/ewf_adapter_renesas_ryz014_internet.c

```diff
@@ -728,7 +727,10 @@ ewf_result ewf_adapter_renesas_ryz014_tcp_receive(ewf_socket_tcp* socket_ptr, ui
             }
             else
             {
-                *buffer_length_ptr = (uint32_t)((*buffer_length_ptr >= recieve_actual_lenght) ? recieve_actual_lenght : *buffer_length_ptr);
```

Comment:
AC6 was optimizing this line and causing `*buffer_length_ptr` to always get set to the passed in value. For example if a buffer length of 1460 was passed in but 13 bytes were actually read it would always get set to 1460.